### DISCUSSION TITLE
How can I selectively enable modules?

### DISCUSSION THREAD

**Discussion by LordUbuntu** [starship/starship#5186] [Q&A]
Right now I have a repository I work in which has something like a hundred different languages in it, which leads to the comical prompt seen here:
![language_madness](https://github.com/starship/starship/assets/32335618/ad894878-c3cc-4b86-ad86-db1d320fcbfe)
I do enjoy the little language emojis and everything, they're very nice! But I also want the prompt to be a bit less noisy. I could try disabling all of that, but where's the fun in that :cry: ?

Is it possible to selectively enable/disable modules based on PWD? Or maybe to limit the language listing to a max number of languages (like maybe 5)?

Thank you.

**Comment by mickimnet** [comment]
You can use the command line to disable / enable modules—for example, `starship config nodejs.disabled true`. This change will be global—meaning will be active in all current shell session.

**Comment by davidkna** [comment]
You can also use negative file matches to solve this, for example the following should hide the nodejs module if there is a `.DISABLE_NODE_MODULE` file:
```toml
[nodejs]
detect_files = ['!.DISABLE_NODE_MODULE', 'package.json', '.node-version']
```

**Reply by LordUbuntu** [reply]
While this is close to what I need, I would prefer that _some_ languages do show, just no so many at once. Isn't there a way to randomly show a limited number of them, but not all of them at once? Perhaps with a custom module somehow?

**Comment by LordUbuntu** [comment]
Thanks for the helpful recommendations! I've decided to embrace the chaos instead of resisting it.
